internal/provider/organization_resource.go

```diff
@@ -0,0 +1,242 @@
+
+type OrganizationResource struct {
+	*CoderdProviderData
+}
+
+// OrganizationResourceModel describes the resource data model.
+type OrganizationResourceModel struct {
+	ID UUID `tfsdk:"id"`
+
+	Name        types.String `tfsdk:"name"`
+	DisplayName types.String `tfsdk:"display_name"`
+	Description types.String `tfsdk:"description"`
+	Icon        types.String `tfsdk:"icon"`
+}
+
+func NewOrganizationResource() resource.Resource {
+	return &OrganizationResource{}
+}
+
+func (r *OrganizationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization"
+}
+
+func (r *OrganizationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "An organization on the Coder deployment",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				CustomType:          UUIDType,
+				Computed:            true,
+				MarkdownDescription: "Organization ID",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the organization.",
+				Required:            true,
+				Validators: []validator.String{
+					codersdkvalidator.Name(),
+				},
+			},
+			"display_name": schema.StringAttribute{
+				MarkdownDescription: "Display name of the organization. Defaults to name.",
+				Computed:            true,
+				Optional:            true,
+				Default:             stringdefault.StaticString(""),
+				Validators: []validator.String{
+					codersdkvalidator.DisplayName(),
+				},
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString(""),
+			},
+			"icon": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Default:  stringdefault.StaticString(""),
+			},
+		},
+	}
+}
+
+func (r *OrganizationResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*CoderdProviderData)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unable to configure provider data",
+			fmt.Sprintf("Expected *CoderdProviderData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.CoderdProviderData = data
+}
+
+func (r *OrganizationResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	// Read Terraform prior state data into the model
+	var data OrganizationResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var org codersdk.Organization
+	var err error
+	if data.ID.IsNull() {
```

Comment:
Your import test supplies a UUID into the name field, but `Organization` and `OrganizationByName` call the same endpoint, so I believe supporting imports by name will always allow you to import by ID? Setting the attributes at the bottom corrects the issue where the name is temporarily the UUID, so I don't think it's bug prone.